homeassistant/components/discovergy/config_flow.py

```diff
@@ -0,0 +1,167 @@
+        """Validate user input and create config entry."""
+        errors = {}
+
+        if user_input:
+            try:
+                await pydiscovergy.Discovergy(
+                    email=user_input[CONF_EMAIL],
+                    password=user_input[CONF_PASSWORD],
+                    app_name=APP_NAME,
+                    httpx_client=get_async_client(self.hass),
+                    authentication=BasicAuth(),
+                ).get_meters()
+
+                result = {"title": user_input[CONF_EMAIL], "data": user_input}
+            except discovergyError.HTTPError:
+                errors["base"] = "cannot_connect"
+            except discovergyError.InvalidLogin:
+                errors["base"] = "invalid_auth"
+            except Exception:  # pylint: disable=broad-except
+                _LOGGER.exception("Unexpected exception")
+                errors["base"] = "unknown"
+            else:
+                if self.existing_entry:
+                    self.hass.config_entries.async_update_entry(
+                        self.existing_entry,
+                        data={
+                            CONF_EMAIL: user_input[CONF_EMAIL],
+                            CONF_PASSWORD: user_input[CONF_PASSWORD],
+                        },
+                    )
+                    await self.hass.config_entries.async_reload(
+                        self.existing_entry.entry_id
+                    )
+                    return self.async_abort(reason="reauth_successful")
+
+                # set unique id to title which is the account email
+                await self.async_set_unique_id(result["title"].lower())
+                self._abort_if_unique_id_configured()
+
+                return self.async_create_entry(
+                    title=result["title"], data=result["data"]
+                )
+
+        return self.async_show_form(
+            step_id=step_id,
+            data_schema=make_schema(),
+            errors=errors,
+        )
+
+    @staticmethod
+    @callback
+    def async_get_options_flow(
+        config_entry: config_entries.ConfigEntry,
+    ) -> config_entries.OptionsFlow:
+        """Create the options flow."""
+        return DiscovergyOptionsFlowHandler(config_entry)
+
+
+class DiscovergyOptionsFlowHandler(config_entries.OptionsFlow):
+    """Handle Discovergy options."""
+
+    def __init__(self, config_entry: config_entries.ConfigEntry) -> None:
+        """Initialize options flow."""
+        self.config_entry = config_entry
+
+    async def async_step_init(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Manage the options."""
+        if user_input is not None:
+            return self.async_create_entry(title="", data=user_input)
+
+        return self.async_show_form(
+            step_id="init",
+            data_schema=vol.Schema(
+                {
+                    vol.Optional(
+                        CONF_TIME_BETWEEN_UPDATE,
```

Comment:
We don't allow scan interval as option. Every integration that supports config entries allows the user to turn off automatic polling and then the user can automate the update interval as needed with the entity service `homeassistant.update_entity`.